packages/kbn-es-archiver/kibana.jsonc

```diff
@@ -1,5 +1,5 @@
 {
-  "type": "shared-common",
+  "type": "shared-server",
```

Comment:
What makes this a "shared-server"?

Reply:
It imports server-only code such as `@kbn/core-saved-objects-server`

Reply:
Since `kbn/es-archiver` is imported only in tests and test-purpose packages, can we set it to `test-helper` type?

Just checked that `kbn-journeys` has `test-helper` as well.

Reply:
ah, sorry. It imports `@kbn/core-saved-objects-server` and then must be defined as `shared-server`, right?

Reply:
This is not confusing at all! hahahaaa
Thanks for chiming in Dima

Reply:
I tried with `"type": "test-helper"` and I'm getting errors when the suites are importing these packages...
_"tests or mocks" code can not be imported from "non-package" code._

Apparently, `test`, `x-pack/test`, and `x-pack/test_serverless` don't have `kibana.jsonc`. 

@elastic/kibana-operations, any recommendations? Do we want to convert the test directions to _packages_?

Reply:
@afharo I think eventually is going to be a good idea to move tests into packages as it will also put them closer to the source code as well. It was our intention from the beginning as well. I think for now is okay for it to stay with shared-server having into account this limitations

Reply:
Then again, I could just do that in my PR lol

Reply:
I tried adding a `kibana.jsonc` to the root dirs of all our test suites (`test`, `x-pack/performance`, `x-pack/test`, and `x-pack/test_serverless`) as `test-helper` and it seems to work.

@mistic, do you think this could be a valid approach?

Reply:
[`10f4b1f` (#188606)](https://github.com/elastic/kibana/pull/188606/commits/10f4b1f59bdfc36edab610a6bb85852df947358e) contains the changes mentioned above.